Core/RetroAchievements.cpp

```diff
@@ -461,6 +469,93 @@ static void login_token_callback(int result, const char *error_message, rc_clien
 	g_isLoggingIn = false;
 }
 
+bool RAIntegrationDirty() {
+#ifdef RC_CLIENT_SUPPORTS_RAINTEGRATION
+	return rc_client_raintegration_has_modifications(g_rcClient);
+#else
+	return false;
+#endif
+}
+
+#ifdef RC_CLIENT_SUPPORTS_RAINTEGRATION
+
+static void raintegration_get_game_name_handler(char* buffer, uint32_t buffer_size, rc_client_t* client)
+{
+	// snprintf(buffer, buffer_size, get_filename(g_loaded_game));
+	// remove_extension(buffer);
+	snprintf(buffer, buffer_size, "%s", g_gamePath.GetFilename().c_str());
+}
+
+static void raintegration_write_memory_handler(uint32_t address, uint8_t* buffer, uint32_t num_bytes, rc_client_t* client) {
+	// convert_retroachievements_address_to_real_address
+	uint32_t realAddress = address + PSP_MEMORY_OFFSET;
+
+	uint8_t *writePtr = Memory::GetPointerWriteRange(address, num_bytes);
+	if (writePtr) {
+		memcpy(writePtr, buffer, num_bytes);
+	}
+}
+
+static void raintegration_event_handler(const rc_client_raintegration_event_t* event, rc_client_t* client) {
+	switch (event->type) {
+	case RC_CLIENT_RAINTEGRATION_EVENT_MENUITEM_CHECKED_CHANGED:
+		// The checked state of one of the menu items has changed and should be reflected in the UI.
+		// Call the handy helper function if the menu was created by rc_client_raintegration_rebuild_submenu.
+		rc_client_raintegration_update_menu_item(client, event->menu_item);
+		break;
+	case RC_CLIENT_RAINTEGRATION_EVENT_PAUSE:
+		// The toolkit has hit a breakpoint and wants to pause the emulator. Do so.
+		Core_EnableStepping(true, "ra_breakpoint");
+		break;
+	case RC_CLIENT_RAINTEGRATION_EVENT_HARDCORE_CHANGED:
+		// Hardcore mode has been changed (either directly by the user, or disabled through the use of the tools).
+		// The frontend doesn't necessarily need to know that this value changed, they can still query it whenever
+		// it's appropriate, but the event lets the frontend do things like enable/disable rewind or cheats.
+		// handle_hardcore_changed();
+		break;
+	default:
+		ERROR_LOG(ACHIEVEMENTS, "Unsupported raintegration event %u\n", event->type);
+		break;
+	}
+}
+
+static void load_integration_callback(int result, const char* error_message, rc_client_t* client, void* userdata) {
+	auto ac = GetI18NCategory(I18NCat::ACHIEVEMENTS);
+
+	// If DLL not present, do nothing. User can still play without the toolkit.
+	switch (result) {
+	case RC_MISSING_VALUE:
+		g_OSD.Show(OSDType::MESSAGE_WARNING, ac->T("RAIntegration is enabled, but RAIntegration-x64.dll was not found."));
+		break;
+	default:
+		{
+			// If not successful, just report the error and bail. Shouldn't happen.
+			if (result != RC_OK) {
+				g_OSD.Show(OSDType::MESSAGE_ERROR, StringFromFormat("RAIntegration init failed: %s", error_message));
+				return;
+			}
```

Comment:
With a switch statement, you should just have lines 534-535 in the default clause and use RC_OK for 538+